cs/src/Management/TunnelManagementClient.cs

```diff
@@ -1241,5 +1242,21 @@ private TunnelPort ConvertTunnelPortForRequest(Tunnel tunnel, TunnelPort tunnelP
                 cancellation);
             return clusterDetails!;
         }
+
+        /// <inheritdoc/>
+        public async Task<bool> CheckNameAvailability(
+            string name,
+            CancellationToken cancellation = default)
+        {
+            Requires.NotNull(name, nameof(name));
+            return await this.SendRequestAsync<bool>(
+                HttpMethod.Get,
+                clusterId: null,
+                TunnelsApiPath + "/" + name + CheckAvailableSubPath,
```

Comment:
URL escape it, or validate that `name` is a valid tunnel name.

Reply:
By URL escape do you mean remove "%"?

Reply:
[Uri.EscapeDataString](https://learn.microsoft.com/en-us/dotnet/api/system.uri.escapedatastring?view=net-7.0)

Reply:
added

Reply:
We should probably add similar escaping in other places where we build the URI path from tunnel ID or name properties, across all the SDK languages.

It doesn't have to block this PR though.